src/cloud/components/onboarding/useGetUserStatus.ts

```diff
@@ -80,48 +62,109 @@ export const getUserStatus = (table: Table): USER_PILOT_USER_STATUS[] => {
+          vector_name: 'writes_mb',
           query: {
-            start: '-30d',
+            range: range ?? '30d',
           },
+        }),
+        getUsage({
+          vector_name: 'storage_gb',
+          query: {
+            range: range ?? '30d',
+          },
+        }),
+        getUsage({
+          vector_name: 'query_count',
+          query: {
+            range: range ?? '30d',
+          },
+        }),
+      ])
+      if (_.every(data, d => d.status === 200)) {
+        const tablesToParse = {}
+        data.forEach(d => {
+          const {table} = fromFlux(d.data)
+          if (table.getColumn('storage_gb')) {
+            tablesToParse['storage_gb'] = table
+          } else if (table.getColumn('writes_mb')) {
+            tablesToParse['writes_mb'] = table
+          } else if (table.getColumn('query_count')) {
+            tablesToParse['query_count'] = table
+          }
         })
-        if (usage?.status === 200) {
-          console.warn(`Usage: ${usage}`)
-          csvToParse = usage.data?.trim().replace(/\r\n/g, '\n')
-          console.warn(`CSVTOPARSE: ${csvToParse}`)
-        }
-      } else {
-        csvToParse = usageStatsCsv
+
+        return tablesToParse
       }
-      const {table} = fromFlux(csvToParse)
-      setUsageDataStates(getUserStatus(table))
     }
-  }, [org?.id])
-
-  useEffect(() => {
-    try {
-      getUserStatusDefinition()
-    } catch (err) {
-      event('cloud.onboarding.set_user_status_failure', {
-        context: JSON.stringify(err),
-      })
-      setUsageDataStates([])
+  } else {
+    return {}
+  }
+}
+
+export const getUserWriteLimitHits = async (): Promise<number> => {
+  try {
+    const table = (await queryUsage('rate_limits', '30d')) as Table
+
+    const fieldNumbers = table.getColumn('_field') as number[]
+    // const limitedWrite = table.getColumn('limited_write', 'number')
+
+    const queryWriteLimitHits = fieldNumbers.reduce((a, b, i) => {
+      if (
+        table.getColumn('_field')[i] === 'limited_write' &&
+        table.getColumn('_value', 'number')[i] > 0
+      ) {
+        return a + table.getColumn('_value', 'number')[i]
+      } else {
+        return a
+      }
+    }, 0)
+
+    return queryWriteLimitHits
+  } catch (err) {
+    console.error(err)
+  }
+}
+
+let hasCalledGetStatus = false
+const handleGetUserStatus = async () => {
+  let usageDataStates = []
+
+  const getUserStatusDefinition = async () => {
+    const tables = await queryUsage()
+    usageDataStates = getUserStatus(tables)
+  }
+
+  try {
+    if (!hasCalledGetStatus && isFlagEnabled('newUsageAPI')) {
+      await getUserStatusDefinition()
+      hasCalledGetStatus = true
     }
-  }, [getUserStatusDefinition])
+  } catch (err) {
+    console.error(err)
+    event('cloud.onboarding.set_user_status_failure', {
+      context: JSON.stringify(err),
+    })
+  }
 
   return {usageDataStates}
 }
 
-export default useGetUserStatus
+export default handleGetUserStatus
```

Comment:
so much less hack, love it! Quick thought though, some of those will return A LOT of data with the `30d`. Do we need to query that long?

Reply:
see response below